pilot/pkg/security/trustdomain/bundle_test.go

```diff
@@ -123,20 +123,57 @@ func TestReplaceTrustDomainAliases(t *testing.T) {
 
 func TestReplaceTrustDomainInPrincipal(t *testing.T) {
 	cases := []struct {
+		name          string
 		trustDomainIn string
 		principal     string
 		out           string
+		expectedError string
 	}{
-		{principal: "spiffe://cluster.local/ns/foo/sa/bar", out: ""},
-		{principal: "sa/test-sa/ns/default", out: ""},
-		{trustDomainIn: "td", principal: "cluster.local/ns/foo/sa/bar", out: "td/ns/foo/sa/bar"},
-		{trustDomainIn: "abc", principal: "xyz/ns/foo/sa/bar", out: "abc/ns/foo/sa/bar"},
+		{
+			name:          "Principal in wrong format with SPIFFE:// prefix",
+			principal:     "spiffe://cluster.local/ns/foo/sa/bar",
+			out:           "",
+			expectedError: "wrong SPIFFE format: spiffe://cluster.local/ns/foo/sa/bar",
+		},
+		{
+			name:          "Principal in wrong format with less components",
+			principal:     "sa/test-sa/ns/default",
+			out:           "",
+			expectedError: "wrong SPIFFE format: sa/test-sa/ns/default",
+		},
+		{
+			name:          "Replace td with domain name in principal",
+			trustDomainIn: "td",
+			principal:     "cluster.local/ns/foo/sa/bar",
+			out:           "td/ns/foo/sa/bar",
+			expectedError: "",
+		},
+		{
+			name:          "Replace td without domain name in principal",
+			trustDomainIn: "abc",
+			principal:     "xyz/ns/foo/sa/bar",
+			out:           "abc/ns/foo/sa/bar",
+			expectedError: "",
+		},
 	}
 
 	for _, c := range cases {
-		got := replaceTrustDomainInPrincipal(c.trustDomainIn, c.principal)
+		got, err := replaceTrustDomainInPrincipal(c.trustDomainIn, c.principal)
+		if err != nil {
+			if c.expectedError == "" {
+				t.Errorf("%s: replace trust domain in principal error: %v", c.name, err)
+			}
+			if got != "" {
```

Comment:
Is this line checking that when err != nil, the got must be empty?

Reply:
Yes.